lib/puppet/provider/package/pear.rb

```diff
@@ -36,7 +36,7 @@ def self.pearlist(hash)
           pearhash[:provider] = :pear
           pearhash
         end
-      end.reject { |p| p.nil? }
+      end.reject(&:nil?)
```

Comment:
`Array#compact` should work, the only difference to this manual solution I could find was that `compact` will only remove actual `nil`, whereas this will work on classes that define a custom `#nil?` (I see no reason why that would be something you want though).